pkg/scheduler/factory/factory_test.go

```diff
@@ -371,35 +336,38 @@ func TestBind(t *testing.T) {
 }
 
 func testBind(binding *v1.Binding, t *testing.T) {
-	handler := utiltesting.FakeHandler{
-		StatusCode:   200,
-		ResponseBody: "",
-		T:            t,
-	}
-	server := httptest.NewServer(&handler)
-	defer server.Close()
-	client := clientset.NewForConfigOrDie(&restclient.Config{Host: server.URL, ContentConfig: restclient.ContentConfig{GroupVersion: &schema.GroupVersion{Group: "", Version: "v1"}}})
+	testPod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: metav1.NamespaceDefault},
+		Spec:       apitesting.V1DeepEqualSafePodSpec(),
+	}
+	client := fake.NewSimpleClientset(&v1.PodList{Items: []v1.Pod{*testPod}})
+
 	b := binder{client}
 
 	if err := b.Bind(binding); err != nil {
 		t.Errorf("Unexpected error: %v", err)
 		return
 	}
+
+	pod := client.CoreV1().Pods(metav1.NamespaceDefault).(*fakeV1.FakePods)
+
+	bind, err := pod.GetBinding("foo")
```

Comment:
Just some side commentary, but the original design of this helper is assuming that the passed in binding object is on a pod named "foo". It makes the test fragile and harder to read. Not something to fix in this PR - I just had to point it out.

Reply:
Emm... it's really easy to make some misunderstanding.